Sources/MongoSwift/Operations/FindOperation.swift

```diff
@@ -0,0 +1,229 @@
+
+    /// The number of documents to skip before returning.
+    public var skip: Int64?
+
+    /// The order in which to return matching documents.
+    public var sort: Document?
+
+    /// If a `CursorType` is provided, indicates whether it is `.tailable` or .`tailableAwait`.
+    private var tailable: Bool?
+
+    // swiftlint:enable redundant_optional_initialization
+
+    /// Convenience initializer allowing any/all parameters to be omitted or optional.
+    public init(
+        allowPartialResults: Bool? = nil,
+        batchSize: Int32? = nil,
+        collation: Document? = nil,
+        comment: String? = nil,
+        cursorType: CursorType? = nil,
+        hint: Hint? = nil,
+        limit: Int64? = nil,
+        max: Document? = nil,
+        maxAwaitTimeMS: Int64? = nil,
+        maxScan: Int64? = nil,
+        maxTimeMS: Int64? = nil,
+        min: Document? = nil,
+        noCursorTimeout: Bool? = nil,
+        projection: Document? = nil,
+        readConcern: ReadConcern? = nil,
+        readPreference: ReadPreference? = nil,
+        returnKey: Bool? = nil,
+        showRecordId: Bool? = nil,
+        skip: Int64? = nil,
+        sort: Document? = nil
+    ) {
+        self.allowPartialResults = allowPartialResults
+        self.batchSize = batchSize
+        self.collation = collation
+        self.comment = comment
+        self.cursorType = cursorType
+        self.hint = hint
+        self.limit = limit
+        self.max = max
+        self.maxAwaitTimeMS = maxAwaitTimeMS
+        self.maxScan = maxScan
+        self.maxTimeMS = maxTimeMS
+        self.min = min
+        self.noCursorTimeout = noCursorTimeout
+        self.projection = projection
+        self.readConcern = readConcern
+        self.readPreference = readPreference
+        self.returnKey = returnKey
+        self.showRecordId = showRecordId
+        self.skip = skip
+        self.sort = sort
+    }
+
+    // Encode everything except `self.readPreference`, because this is sent to libmongoc separately
+    private enum CodingKeys: String, CodingKey {
+        case allowPartialResults, awaitData, batchSize, collation, comment, hint, limit, max, maxAwaitTimeMS,
+            maxScan, maxTimeMS, min, noCursorTimeout, projection, readConcern, returnKey, showRecordId, tailable, skip,
+            sort
+    }
+}
+
+/// An operation corresponding to a "find" command on a collection.
+internal struct FindOperation<CollectionType: Codable>: Operation {
+    private let collection: MongoCollection<CollectionType>
+    private let filter: Document
+    private let options: FindOptions?
+
+    internal init(collection: MongoCollection<CollectionType>, filter: Document, options: FindOptions?) {
+        self.collection = collection
+        self.filter = filter
+        self.options = options
+    }
+
+    internal func execute(
+        using connection: Connection,
+        session: ClientSession?
+    )
+        throws -> MongoCursor<CollectionType> {
```

Comment:
I feel like this line could be on the same as the closing paren, does SwiftFormat not put it there? (If not, does it complain if you put it there?)

Reply:
it does not complain. I maybe did this myself before I ran the formatter.... updated to be on one line.